odxtools/cli/browse.py

```diff
@@ -31,28 +34,38 @@ def _convert_string_to_odx_type(string_value: str, odx_type: DataType):
         return odx_type.from_string(string_value)
 
 
-def _convert_string_to_bytes(string_value):
+def _convert_string_to_bytes(string_value: str) -> bytes:
     if all(len(x) <= 2 for x in string_value.split(" ")):
         return bytes(int(x, 16) for x in string_value.split(" ") if len(x) > 0)
     else:
         return int(string_value, 16).to_bytes((int(string_value, 16).bit_length() + 7) // 8, "big")
 
 
-def _validate_string_value(input, parameter):
-    if parameter.is_optional and input == "":
+def _validate_string_value(input: str, parameter: Parameter) -> bool:
+    if not parameter.is_required and input == "":
         return True
     elif isinstance(parameter, ParameterWithDOP):
         try:
-            val = _convert_string_to_odx_type(input, parameter.physical_type.base_data_type)
+            phys_type = odxrequire(parameter.physical_type)
+            val = _convert_string_to_odx_type(input, phys_type.base_data_type)
         except:  # noqa: E722
             return False
-        return parameter.dop.is_valid_physical_value(val)
+        dop = parameter.dop
+        if isinstance(dop, DataObjectProperty):
+            return dop.is_valid_physical_value(val)
```

Comment:
is_valid_physical_value, is defined in base dop, right? so it's the implementation of dop sub class that should raise the not implemented error

Reply:
Just checked: it only available for simple DOPs (i.e., `DataObjectProperty`, not `DopBase`) but you're right: it should probably me moved (I'll deal with it on Monday...)

Reply:
okay, fixed: 7e4a516cec685a4bb25b1029cfed62070c8bf863. thanks for noticing!